blackjax/inference/nuts.py

```diff
@@ -0,0 +1,115 @@
+from typing import NamedTuple
+import jax
+import jax.numpy as jnp
+
+integrator = None
+max_tree_depth = 0
+rng_key = 0
+step_size = 0
+
+"""
+TODO:
+- U-turn checks;
```

Comment:
cf https://discourse.mc-stan.org/t/nuts-misses-u-turns-runs-in-circles-until-max-treedepth/9727/55